### DISCUSSION TITLE
Reconsider Magnifier and other addons that would use third-party APIs

### DISCUSSION THREAD

**Discussion by mxmou** [ScratchAddons/ScratchAddons#2878] [Discussion]
About three weeks ago, the creator of [Magnifier](https://magnifier.potatophant.net/browse), a website for viewing and reacting to Scratch comments, made a [fork](https://github.com/CluckCluckChicken/ScratchAddons) of Scratch Addons with the intention of creating a PR in this repo so that Scratch Addons users could easily react to comments.

They said it was rejected in the Discord server for reasons related to third party servers. I don't know the exact reason, but I think it should be reconsidered. Even if we eventually have our own servers, integration with existing Scratch-related websites should be one of SA's goals. We already have addons that use ocular and ScratchDB, and a link to the privacy policy should be enough to allow including any third-party service.

Making a separate extension just for Magnifier is not an alternative: it would be hard for Scratchers to discover it, which is exactly why Scratch Addons was created. From the original suggestion:

> It's a fun project (because of this many addons in an extension architecture) and it would be way easier for new userscripts, userstyles or extensions made by Scratchers to be available immediately to existing extension users.

**Comment by TheColaber** [comment]
I'll bring some convo to you:
Latest comments by W_L:
> the idea is good
but we should explore it after we add centrally planned online status, current activity, etc.
so that we can enable it by default

> in case I wasn't clear enough, your addon probably won't get accepted :stuck_out_tongue:
or it might if you really want to but I wouldn't expect many users

> bruh magnifier wasn't officially rejected
i just said it *probably* wont get in

**Reply by mxmou** [reply]
So the real reason is "we won't add it until we can enable it by default"? Can't addons become enabled by default later?

**Comment by apple502j** [comment]
It depends on the addon.

Addons that work opt-in are welcome. Addons that have to be opt-out aren't.
Also we'd like to have close contact with the developers of the external API.

Is the developer of Magnifier in our discord?

**Reply by mxmou** [reply]
> It depends on the addon.
> 
> Addons that work opt-in are welcome. Addons that have to be opt-out aren't.
> Also we'd like to have close contact with the developers of the external API.
> 
> Is the developer of Magnifier in our discord?

It would probably work very similarly to ocular: not enabled by default and the description tells where the data is from.

I think they asked about it on Discord and W_L said it probably won't be accepted.

**Reply by WorldLanguages** [reply]
I think the main reasoning I had at the moment was about the usefulness of emoji reactions in comments. The only way for features like that to be useful is for them to be enabled by default, and that can't be the case for Magnifier. Ocular forum post reactions are similar but not the same. You could react to forum posts through the Ocular web client before it was an addon, and forum features are a niche anyway, I bet most of our users don't use the forums.

**Reply by WorldLanguages** [reply]
I guess my point at the moment was that we often accept pretty useless addons, as well as addons that use external APIs, but we probably wouldn't accept a combination of both.

**Reply by mxmou** [reply]
Magnifier also has a web client, so I don't think it's useless if it isn't enabled by default.

**Reply by WorldLanguages** [reply]
I don't think a web client for viewing the forums and for reading project/profile/studio comments are comparable. The forums are all about reading the posts themselves, and users don't post as often as in comments.

**Reply by Hans5958** [reply]
> I guess my point at the moment was that we often accept pretty useless addons, as well as addons that use external APIs, but we probably wouldn't accept a combination of both.

I wonder why we should limit what gets in and what gets out. My creation, "Curator links," is next to useless, yet it is added. **I don't think we should start doing this gatekeeping fiasco.** 

(with an exception to ethical reasons such as the invite all followers button, which I still pity about because we are not giving any warranty and liability to them, yet everyone seemed to be scared)

**Reply by mxmou** [reply]
> I wonder why we should limit what gets in and what gets out. My creation, "Curator links," is next to useless, yet it is added. **I don't think we should start doing this gatekeeping fiasco.**

If at least some users (in this case those are members of [this studio](https://scratch.mit.edu/studios/3688309/comments/)) consider an addon useful, I think it should be added.

**Reply by WorldLanguages** [reply]
> with an exception to ethical reasons such as the invite all followers button

Scratch ratelimits studio invites so it wouldn't be very practical anyway

> in this case those are members of [this studio](https://scratch.mit.edu/studios/3688309/comments/)) consider an addon useful

Are we talking about magnifier?

**Reply by mxmou** [reply]
> > in this case those are members of [this studio](https://scratch.mit.edu/studios/3688309/comments/)) consider an addon useful
> 
> Are we talking about magnifier?

Yes - many members of the studio know about and use Scratch Addons. Reacting to comments elsewhere is less useful because few people will see the reactions.

**Reply by Hans5958** [reply]
> Scratch ratelimits studio invites so it wouldn't be very practical anyway

Well, considering that they have patience of clicking toms of users, I'm sure that they won't mind being waiting for a while.

**Comment by davwat** [comment]
this is a good idea

**Reply by davwat** [reply]
not biased
